### PR TITLE
SCRD-4272 Add Gerrit Change ID param to openstack-ardana job

### PR DESCRIPTION
**Pull request by cmurphy**
This change adds the ability to point the openstack-ardana Jenkins job
at an open change proposal on gerrit.suse.provo.cloud, build it on IBS,
and use the resulting repository as an update repository for Ardana
integration tests.

It adds a new job, cloud-ardana-testbuild-gerrit, which uses the
openstack-trackupstream node label since it requires IBS credentials.
This job uses a python script, build_test_package.py, which borrows
heavily from the rpm-packaging project's createproject.py script to
create a test project in IBS, copy a package into it, point the source
service at a locally updated version of the source, and build the
package. The repository has the publish flag set so it can be used by
the integration test as an additional repository on top of the regular
Cloud repository that is mirrored in Provo. By default, it creates a
subproject of Devel:Cloud:8 to test in.

It adds a conditional build step to the openstack-ardana job that
triggers the new job if the gerrit_change_id parameter is set. For now,
users can manually set this to test a change they want feedback on, but
in the future this could be automatically set by a trigger that reacts
to new changes in Gerrit.

TODO:
  - ~~Add IBS project cleanup~~
  - ~~Fill out repo->package map file for all Ardana repos~~
  - ~~Add ability to check out more than one change proposal~~

Notes on the project map: it excludes:

- the "ardana" package which isn't directly tied to any source, even
   though it should be using the ardana repository
 - python-ardana-packager which comes from ardana-dev-tools and whose
   source service uses the PyPI tarball instead of git
 - ardana-installer-server and ardana-installer-ui which are developed
   on GitHub
 - python-swiftlm which uses a tarball instead of the source service
 - python-ardana-configurationprocessor which uses a tarball instead of
   the source service
 - python-glance-check which uses a tarball instead of the source
   service
 - manila-ansible, which we don't package

### CODE REVIEW

File: scripts/jenkins/ardana/gerrit/build_test_package.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            service_file.seek(0)
+            service_file.write(service_def)
+            service_file.truncate()
+        sh.osc('rm', glob.glob('%s*.obscpio' % package_name))
+        sh.osc('service', 'disabledrun')
+        sh.osc('add', glob.glob('%s*.obscpio' % package_name))
+        sh.osc('commit', '-m', 'Testing change %s' % change.id)
+        # Wait for build to be scheduled
+        while 'unknown' in sh.osc('results'):
+            sleep(3)
```
Comment (bmwiedemann) [inline scripts/jenkins/ardana/gerrit/build_test_package.py]: This should be `time.sleep(3)`

and flake8 (via travis-ci) has some other complaints about this file that will need to be addressed.

**Comment by cmurphy** [comment]
Fixed the travis and codeclimate issues.

**Comment by stefannica** [comment]
I rebased the changes, to include latest openstack-ardana fixes.

**Comment by brett060102** [comment]
I may have missed it in the code, but how are you dealing with the issue that change IDs may have more then on review associated with them?

See: https://gerrit.suse.provo.cloud/#/q/Ie6ad5bd4923a45296b33028da8288068c39cdf74 for example.

**Comment by cmurphy** [comment]
[**@brett060102**](https://github.com/brett060102) commented on [Jul 30, 2018, 11:06 PM GMT+2](https://github.com/SUSE-Cloud/automation/pull/2588#issuecomment-409010681 "2018-07-30T21:06:56Z - Replied by Github Reply Comments"):
> I may have missed it in the code, but how are you dealing with the issue that change IDs may have more then on review associated with them?
> 
> See: https://gerrit.suse.provo.cloud/#/q/Ie6ad5bd4923a45296b33028da8288068c39cdf74 for example.

I've been assuming this would use the 4-digit number in the URL of the review, e.g. https://gerrit.suse.provo.cloud/#/c/4482/ , not the change-ID footer in the commit message. The 4-digit ID should be unique, but I can see how the name of the variable is ambiguous. Suggestion on what it should be called?

**Pull request merged**